### PR TITLE
Removed no-module configuration option from libcrypto source build

### PR DESCRIPTION
**Pull request by pranavrth**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,11 @@ librdkafka v2.1.0 is a feature release:
 
 ## Fixes
 
+### OpenSSL fixes
+
+ * Fixed OpenSSL static build not able to include new modules like FIPS
```
Comment (emasab) [inline CHANGELOG.md:37]: better to say "to use external modules like FIPS"

**Review by emasab — APPROVED** [review]
LGTM, just that changelog comment to address. This allows to use the FIPS module with the statically linked OpenSSL 3.0 apart than by dynamically linking OpenSSL compiled along with FIPS module.

**Comment by pranavrth** [comment]
> This allows to use the FIPS module with the statically linked OpenSSL 3.0 apart than by dynamically linking OpenSSL compiled along with FIPS module.

Didn't understand this. Do I need to add this as well in the Changelog apart from what I have added?

**Comment by pranavrth** [comment]
> > This allows to use the FIPS module with the statically linked OpenSSL 3.0 apart than by dynamically linking OpenSSL compiled along with FIPS module.
> 
> Didn't understand this. Do I need to add this as well in the Changelog apart from what I have added?

As discussed over slack, will be adding regarding this in the introduction.

**Comment by navidsh** [comment]
Hi @pranavrth and @emasab, do you happen to know the answer to https://github.com/confluentinc/librdkafka/discussions/4343?

**Pull request merged**